src/Common/TimerDescriptor.cpp

```diff
@@ -75,10 +76,22 @@ void TimerDescriptor::drain() const
     /// or since the last successful read(2), then the buffer given to read(2) returns an unsigned 8-byte integer (uint64_t)
     /// containing the number of expirations that have occurred.
     /// (The returned value is in host byte order—that is, the native byte order for integers on the host machine.)
+
+    /// Due to a bug in Linux Kernel, reading from timerfd in non-blocking mode can be still blocking.
```

Comment:
do you have a link to this bug? what I have found is this:
https://patchwork.kernel.org/project/linux-fsdevel/patch/b4059ed0-5567-44e7-95f7-f7e4b227501c@kernel.dk/
but we avoiding this by breaking the loop on `EAGAIN`

Reply:
No, I didn't find it.